src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -208,38 +203,30 @@ Map<String, CreateExperimentAPIObject> getExperimentMap(DataSourceMetadataInfo m
-                                    rs.setThreshold(CREATE_EXPERIMENT_CONFIG_BEAN.getThreshold());
-                                    createExperimentAPIObject.setRecommendationSettings(rs);
-                                    TrialSettings trialSettings = new TrialSettings();
-                                    trialSettings.setMeasurement_durationMinutes(CREATE_EXPERIMENT_CONFIG_BEAN.getMeasurementDurationStr());
-                                    createExperimentAPIObject.setTrialSettings(trialSettings);
-                                    List<KruizeObject> kruizeExpList = new ArrayList<>();
-
-                                    createExperimentAPIObject.setExperiment_id(Utils.generateID(createExperimentAPIObject.toString()));
-                                    createExperimentAPIObject.setStatus(AnalyzerConstants.ExperimentStatus.IN_PROGRESS);
-                                    createExperimentAPIObject.setExperimentType(AnalyzerConstants.ExperimentTypes.CONTAINER_EXPERIMENT);
-                                    createExperimentAPIObjectMap.put(experiment_name, createExperimentAPIObject);
+                                    // Experiment name - dynamically constructed
+                                    String experiment_name = this.bulkInput.getDatasource() + "|" + dsc.getDataSourceClusterName() + "|"
+                                            + namespace.getDataSourceNamespaceName() + "|" + dsw.getDataSourceWorkloadName() + "("
+                                            + dsw.getDataSourceWorkloadType() + ")" + "|" + dc.getDataSourceContainerName();
+                                    // create JSON to be passed in the createExperimentAPI
+                                    List<CreateExperimentAPIObject> createExperimentAPIObjectList = new ArrayList<>();
+                                    String createExpJSON = prepareCreateExperimentJSONInput(dc, dsc, dsw, namespace,
+                                            experiment_name, createExperimentAPIObjectList);
+                                    // send request to createExperiment API for experiment creation
+                                    GenericRestApiClient apiClient = new GenericRestApiClient(datasource);
+                                    apiClient.setBaseURL(KruizeDeploymentInfo.experiments_url);
+                                    int responseCode;
+                                    try {
+                                        responseCode = apiClient.callKruizeAPI(createExpJSON);
+                                        LOGGER.debug("API Response code: {}", responseCode);
+                                    } catch (Exception | FetchMetricsError e) {
+                                        e.printStackTrace();
+                                        continue;
+                                    }
+                                    // if the experiment is successfully created, add it in the map
```

Comment:
if duplicate experiments then also add it to createExperimentAPIObject
if it failed then it should be mentioned in JOB details

Reply:
here failed one should get handled
jobData.getData().getExperiments().setFailedExperiments